packages/RichTextEditing-Core.package/DocumentReadStream.class/instance/endOfStructure.st

```diff
@@ -0,0 +1,11 @@
+stop conditions
+endOfStructure
+	| currentStructures nextStructures diff |
+	
+	self position + 1 > self readLimit ifTrue: [^ nil].
+	currentStructures := self collection attributesAt: self position.
+	nextStructures := self collection attributesAt: self position + 1.
+	diff := currentStructures difference: nextStructures.
```

Comment:
Create a separate method for this (`structureDifferenceBetween: firstPosition and: secondPosition` or `structureDifferenceToNext`?) as it is used in `endOfStucture` and `beginningOfStructure` and maybe somewhere else in the future.